openedx_cas/backends.py

```diff
@@ -1,41 +1,135 @@
     stored on extra_data field. The setting name is created following the
     pattern SOCIAL_AUTH_<uppercase current backend name>_EXTRA_DATA.
 
-    access_token is always stored.
-
     URLs settings:
-        AUTHORIZATION_URL       Authorization service url
-        ACCESS_TOKEN_URL        Access token URL
+        CAS_SERVICE_URL:
+            Our authorization service url.
+        CAS_CLIENT_VERSION:
+            Version of client protocol.
+            Currently V2 and V3 versions are supported, not sure how V1 will perform
+        CAS_SERVER_URL:
+            Server URL against the user is authenticated
     """
+
     name = "centralized-auth-service"
 
     def auth_url(self):
         """Get the URL to which we must redirect in order to
         authenticate the user:
 
-        Redirect /CAS-provider-domain/
+        Redirect /CAS-service-domain/
         """
-        raise NotImplementedError("Not implemented yet")
+        if not getattr(settings, "CAS_SERVER_URL", None):
+            raise AuthMissingParameter(self, "CAS_SERVER_URL")
+        if not getattr(settings, "CAS_SERVICE_URL", None):
+            raise AuthMissingParameter(self, "CAS_SERVICE_URL")
+        return f"{settings.CAS_SERVER_URL}?service={urllib.parse.quote(settings.CAS_SERVICE_URL)}"
 
     def auth_complete(self, *args, **kwargs):
         """Completes login process, must return user instance.
 
-        We could use the implementation of:
-        https://github.com/django-cas-ng/django-cas-ng/blob/master/django_cas_ng/backends.py#L22
-        And refactor it as we see fit according our use case.
-
         Request:
-        GET
-	    https://CAS-provider-domain/auth/complete/?ticket=ticket
+        GET https://CAS-service-domain/auth/complete/?ticket=ticket
+        :returns: [User] Authenticated User object or None if authenticate failed
+        """
+
+        request = kwargs["request"]
+        ticket = request.GET["ticket"]
+        response = self.cas_validation(request, ticket, settings.CAS_SERVICE_URL)
+        kwargs.update({"response": response, "backend": self})
+
+        logger.info(f"User {response} logged in with {settings.CAS_SERVER_URL}")
+        
+        return self.strategy.authenticate(*args, **kwargs)
+
+    def cas_validation(self, request, ticket, service):
+        """Performs the CAS validation process
+        
+        Overrides CASBackend.cas_validation to make it compatible with third_party_auth app
+        
+        Get the CAS_CLIENT based on the settings.CAS_VERSION and verify the ticket
+        which returns the authentication information of the user.
+
+        Parameters:
+            request: HttpRequest
+            ticket: str containing the ticket key
+            service: setting used 
+        Returns:
+            is successful:
+                return a dict containing the username
+            otherwhise:
+                raise AuthFailed exception or None when the username attribute is changed
```

Comment:
```suggestion
        Returns:
            - A dictionary containing the username when the validation flow succeeds
            - None when the username attribute is changed
            
      Raises:
                AuthFailed exception when...
```
The indentation is all over the place but you get the idea